### PR TITLE
Add Arch Linux image to run tests with gcc7 and arm-none-eabi-gcc7

### PR DESCRIPTION
**Pull request by jlecoeur**
The build on newer gcc versions is broken. Here is an Arch image with gcc7 and arm-none-eabi-gcc7 installed. 
I suggest to run CI on this image for posix_sitl_default, px4fmu-v2_default and px4fmu-v5_default.
@dagar

### CODE REVIEW

**Comment by dagar** [comment]
Seems fine to me if there's nothing more official. You don't want to use the latest pritunl/archlinux? 

In the future I was thinking of centralizing the instructions from the dev guide (https://dev.px4.io/en/setup/dev_env_linux_boutique.html) in a repo, or possibly the firmware itself. Then have the docker build use them directly. This would help keep just about everyone on the same page and catch environment setup problems as they break.

As a side note, in the other containers I forced everything into a single RUN to minimize the size as travis-ci and semapahoreci were pulling every single build. This can relax a bit with the new jenkins setup that will usually already have the images.

**Comment by jlecoeur** [comment]
I missed the official image https://hub.docker.com/r/archlinux/base/, let's use that one instead.

The official image has no tags so that answers your question! 
I did not use latest because Arch is a rolling release, so the px4 arch image will be different at each build even if the dockerfile was not changed. The good side of using latest is we will know rapidly when the Arch build is broken!

It's a good idea to centralize the install scripts. Doing this Dockerfile I noticed that the devs guide is not quite complete for Arch.

I did not know that multiple RUN increased the size, I'll fix 

About the size, Debian images are smaller than Ubuntu's, what about moving to Debian?

**Comment by hamishwillee** [comment]
I ran out of time to create ArchLinux and CentOS scripts. When you have working versions, can you point me to them so I can update the docs? I was thinking of installing apt-get rather than using yaourt so that the instructions could be exactly the same as for Ubuntu - thoughts?

And yes, would be great if the scripts for ubuntu (and others) were in the firmware repo. My only requirement is that it should be possible for a user to take those scripts and run them on a fresh install of OS "whatever" and have the environment set up. If the scripts are smart enough to also cope with existing partial installs etc, then that is a bonus.

**Comment by jlecoeur** [comment]
@dagar I made all the changes
@hamishwillee I added the install script in *docker/px4-dev/scripts/*. Feel free to move it to Firmware and/or DevGuide.

**Comment by hamishwillee** [comment]
Nice. No plans to also install the simulator environment in those scripts?

**Comment by jlecoeur** [comment]
For this docker image, no. The goal is to check compilation on several distros, SITL will be run only on ubuntu as far as I know.
For the dev guide, probably yes. Do you want several install scripts like the ubuntu scripts in dev guide?

**Comment by hamishwillee** [comment]
> For the dev guide, probably yes. Do you want several install scripts like the ubuntu scripts in dev guide?

Ideally. I don't mind so much about ArchLinux - we explicitly state that Ubuntu is our main distro. Even what you have here is a vast improvement on what we have. FYI note that the scripts currently download (wget) "base" scripts - so we can use your docker for the base should we wish and benefit from it being tested even if the sim parts aren't

File: docker/px4-dev/scripts/archlinux_install_script.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# From clean distribution of ArchLinux, installs 
+# all required packages for PX4 development on 
+# posix and nuttx targets
+
+# Install required PX4 packages
+yes | pacman -Sy --noconfirm \
+		base-devel \
+		make \
+		cmake \
+		git-core \
```
Comment (hamishwillee) [inline docker/px4-dev/scripts/archlinux_install_script.sh]: Is git-core valid - for ubuntu at least @dagar suggested using `git`

File: docker/px4-dev/scripts/archlinux_install_script.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
```
Comment (hamishwillee) [inline docker/px4-dev/scripts/archlinux_install_script.sh]: @jlecoeur Original docs additionally installed these: lib32-glibc python-pyserial python-numpy  toml - are they not needed on archlinux?

File: docker/px4-dev/scripts/archlinux_install_script.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+# From clean distribution of ArchLinux, installs 
+# all required packages for PX4 development on 
+# posix and nuttx targets
+
+# Install required PX4 packages
```
Comment (hamishwillee) [inline docker/px4-dev/scripts/archlinux_install_script.sh]: What about ninja?

File: docker/px4-dev/scripts/archlinux_install_script.sh (hunk 2 above)
Comment (jlecoeur) [reply docker/px4-dev/scripts/archlinux_install_script.sh]: @hamishwillee not for what I tested at least.
I has been in the doc for a long time: https://github.com/PX4/Devguide/commit/b2131f18f35d5cbd87182f21b2d5681c94937877#diff-1e96ef0fe3c2abb013c67b2245857406
Maybe it is not required anymore?

Comment (hamishwillee) [reply docker/px4-dev/scripts/archlinux_install_script.sh]: Thanks. @dagar , any idea whether  lib32-glibc  is needed in builds? / What is it used for?

Comment (dagar) [reply docker/px4-dev/scripts/archlinux_install_script.sh]: Probably for the 32bit version of the arm toolchain. The next time we update it will be 64bit linux. 

image
